src/chunks/vocab/subword.rs

```diff
@@ -27,11 +27,11 @@ pub type ExplicitSubwordVocab = SubwordVocab<ExplicitIndexer>;
 /// Vocabulary with subword units.
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct SubwordVocab<I> {
-    indexer: I,
-    indices: HashMap<String, usize>,
-    words: Vec<String>,
-    min_n: u32,
-    max_n: u32,
+    pub indexer: I,
```

Comment:
The struct fields are implementation details and should be kept hidden, so that we can change the implementation in the future without changing the API.

It would be best to offer a constructor that takes: an indexer, the words, min_n, and max_n (`indices` can be constructed from `words`).

Reply:
Thanks, I'll get to work on this and submit another PR (hopefully by next week).